lib/xnvme_dev.c

```diff
@@ -426,48 +427,79 @@ xnvme_dev_pr(const struct xnvme_dev *dev, int opts)
 const struct xnvme_geo *
 xnvme_dev_get_geo(const struct xnvme_dev *dev)
 {
+	struct xnvme_dev *dropped = (struct xnvme_dev *)dev;
+
+	if (!dev->is_derived) {
+		dropped->is_derived = xnvme_dev_derive_geo(dropped);
```

Comment:
I get that this wouldn't be possible if `is_derived` was a bool, but perhaps that is for the better.
If this fails, then a "truthy" value will be stored in `is_derived` even though it is not actually derived.
In general I think the error handling in this commit, needs a second pass.

Reply:
... renamed as well to better reflect what they actually represented... and **attempt** has been made...